uber/templates/signups/shifts.html

```diff
@@ -158,18 +157,13 @@
                 element.find('.fc-title').append('</br>' + buttonTag);
             }
         });
-        {% if 'schedule' in c.QUERY_STRING %}
-            $("#shift_cal").fullCalendar( 'changeView','listWeek');
-        {% else %}
-            // if opened with tgt_date, go to date in calendar
+        {% if view %}
+            // Keep the view they were on when they added or dropped a shift
+            $('#shift_cal').fullCalendar('changeView', '{{ view }}');
+            {% if start %}
+                $('#shift_cal').fullCalendar('gotoDate', '{{ start }}');
```

Comment:
By the way, `'changeView'` (the function above this line) is supposed to be able to go to just any date when you change the view. Unfortunately, it didn't work for some reason, so I used gotoDate instead.